src/adagrams.js

```diff
@@ -1,15 +1,125 @@
+
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  var LETTERQUANTITY = {
+    letters: [
+      {letter: "A", quantity: 9},
+      {letter: "B", quantity: 2},
+      {letter: "C", quantity: 2},
+      {letter: "D", quantity: 4},
+      {letter: "E", quantity: 12},
+      {letter: "F", quantity: 2},
+      {letter: "G", quantity: 3},
+      {letter: "H", quantity: 2},
+      {letter: "I", quantity: 9},
+      {letter: "J", quantity: 1},
+      {letter: "K", quantity: 1},
+      {letter: "L", quantity: 4},
+      {letter: "M", quantity: 2},
+      {letter: "N", quantity: 6},
+      {letter: "O", quantity: 8},
+      {letter: "P", quantity: 2},
+      {letter: "Q", quantity: 1},
+      {letter: "R", quantity: 6},
+      {letter: "S", quantity: 4},
+      {letter: "T", quantity: 6},
+      {letter: "U", quantity: 4},
+      {letter: "V", quantity: 2},
+      {letter: "W", quantity: 2},
+      {letter: "X", quantity: 1},
+      {letter: "Y", quantity: 2},
+      {letter: "Z", quantity: 1},
+    ]
+  };
+
+  var letterBank = [];
+  let counter = 10;
+  let letters = LETTERQUANTITY.letters;
+  while (counter > 0) {
+    let randI = Math.trunc(Math.random()* letters.length);
+
+    if (letters[randI].quantity === 0) {
+      continue;
+    }
+
+    let quantity = letters[randI].quantity - 1;
+    letters[randI].quantity = (quantity);
+    letterBank.push(letters[randI].letter);
+    counter -= 1;
+  }
+  return letterBank;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+
+  for (let letter of input) {
+    if (lettersInHand.includes(letter.toUpperCase())) {
+      let i = lettersInHand.indexOf(letter.toUpperCase());
+      lettersInHand.splice(i, 1);
+    } else {
+      return false;
+    }
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+
+  let letterPoints = {
+    "A":1, "B":3, "C":3, "D":2, "E":1, "F":4, "G":2, "H":4, "I":1, "J":8, "K":5, "L":1,
+    "M":3, "N":1, "O":1, "P":3, "Q":10, "R":1, "S":1, "T":1, "U":1, "V":4, "W":4, "X":8,
+    "Y":4, "Z":10
+    };
+
+  let totalPoints = 0;
+
+  for (let letter of word) {
+    if (Object.keys(letterPoints).includes(letter.toUpperCase())) {
+      totalPoints += letterPoints[letter.toUpperCase()];
+    }
+  }
+
+  if (word.length >= 7 && word.length <= 10) {
+    totalPoints += 8;
+  }
+
+  return totalPoints;
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
+  let scoreDict = {};
+
+  for (let word of words) {
+    scoreDict[word] = scoreWord(word);
+  }
```

Comment:
Nice use of for..of. We can actually have `word` be a `const` here because it's not reassigned inside the loop.